kernel/src/table_changes/scan.rs

```diff
@@ -314,7 +315,14 @@ fn read_scan_file(
         // trying to return a captured variable. We're going to reassign `selection_vector`
         // to `rest` in a moment anyway
         let mut sv = selection_vector.take();
-        let rest = split_vector(sv.as_mut(), len, None);
+
+        // If this is a resolve_dv_pair, we extend with false because we only want to select rows
```

Comment:
I feel like `resolve_dv_pair` is determining if this is a _selection_ vs _deletion_ vector. That's confusing. Can we just make it a selection vector always? 

I get that up-front we might not be able to, but could `ResolvedCdfScanFile` make `selection_vector` private (and maybe call it something else since it can be both), and then have a `selection_vector()` method that "does the right thing".

Reply:
> I feel like resolve_dv_pair is determining if this is a selection vs deletion vector.

Not quite. Both are selection vectors, its just that the way they're extended is different. 

Consider data with 5 rows. Suppose in case A we have a deletion treemap with 1 set `Treemap_d(1)`. This is converted into a selection vector `[1, 0, 1]`. (this is what happens in `selection_treemap_to_bools`).
For 5 rows this *really* translates into a selection vector: `[1, 0, 1, 1, 1]`, since nothing else was deleted, we select all the rows that are greater than sv.length.

Consider the same 5 data rows, and we resolved the selection map to be Treemap_s(1). This is converted into a selection vector `[0, 1]`. But this selection vector *only* wants to select element 1. Nothing else. So  this really translates to selection vector `[0, 1, 0, 0, 0]`.

Reply:
Ahh, okay. I'd still vote to hide this inside an `extend` or similar method on `ResolvedCdfScanFile` so others don't have to write this code.